activerecord/lib/active_record/connection_adapters/postgresql/migration/compatibility.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      class Migration
+        module Compatibility # :nodoc: all
+          def self.find(version)
+            version = version.to_s
+            name = "V#{version.tr('.', '_')}"
+            if const_defined?(name)
+              return const_get(name)
+            else
+              ActiveRecord::Migration::Compatibility.find(version)
+            end
+          end
+
+          class V5_1 < ActiveRecord::Migration::Compatibility::V5_1
+            def change_column(table_name, column_name, type, options = {})
+              clear_cache!
+              sql = connection.send(:change_column_sql, table_name, column_name, type, options)
+              execute "ALTER TABLE #{quote_table_name(table_name)} #{sql}"
+              change_column_default(table_name, column_name, options[:default]) if options.key?(:default)
+              change_column_null(table_name, column_name, options[:null], options[:default]) if options.key?(:null)
+              change_column_comment(table_name, column_name, options[:comment]) if options.key?(:comment)
+            end
+          end
+
+          class V5_0 < ActiveRecord::Migration::Compatibility::V5_0
```

Comment:
This is where things get tricky: this inheritance will bypass the above `change_column`, which should apply.

It feels like we really want multiple inheritance, which in Ruby means we need modules instead of [some] classes.

To that end, I suggest we make these adapter-specific parts modules -- then when we ask for "PG 5.0", we would get a `Class.new(ActiveRecord::Migration::Compatibility::V5_0) { include PG::V5_1; include PG::V5_0 }`... maybe?

It's unfortunate that the PG::V5_1 would be built on the base V5_0... the design is supposed to ensure newer versions know nothing about (and are unaffected by) older ones. A more radical redesign might produce: `Class.new(Migration) { include Base::V5_1; include PG::V5_1; include Base::V5_0; include PG::V5_0 }` (i.e., making everything modules)

Reply:
Thanks for the suggestion. right. 

`ActiveRecord::ConnectionAdapters::PostgreSQL::Migration::Compatibility::V5_1` 
is not any of ancestors. Let me think about an option to use module based one.

```ruby
From: /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/postgresql/migration/compatibility.rb @ line 30 :

    25:               change_column_comment(table_name, column_name, options[:comment]) if options.key?(:comment)
    26:             end
    27:           end
    28:
    29:           class V5_0 < ActiveRecord::Migration::Compatibility::V5_0
 => 30:             binding.irb
    31:             pp self.ancestors
    32:             def create_table(table_name, options = {})
    33:               if options[:id] == :uuid && !options.key?(:default)
    34:                 options[:default] = "uuid_generate_v4()"
    35:               end

irb(ActiveRecord::ConnectionAdapters::PostgreSQL::Migration::Compatibility::V5_0):001:0> pp self.ancestors
[ActiveRecord::ConnectionAdapters::PostgreSQL::Migration::Compatibility::V5_0,
 ActiveRecord::Migration::Compatibility::V5_0,
 ActiveRecord::Migration::Compatibility::V5_1,
 ActiveRecord::Migration::Compatibility::V5_2,
 ActiveRecord::Migration::Current,
 ActiveRecord::Migration,
 ActiveSupport::ToJsonWithActiveSupportEncoder,
 Object,
 ActiveSupport::Tryable,
 ActiveSupport::Dependencies::Loadable,
 JSON::Ext::Generator::GeneratorMethods::Object,
 Kernel,
 BasicObject]
=> [ActiveRecord::ConnectionAdapters::PostgreSQL::Migration::Compatibility::V5_0, ActiveRecord::Migration::Compatibility::V5_0, ActiveRecord::Migration::Compatibility::V5_1, ActiveRecord::Migration::Compatibility::V5_2, ActiveRecord::Migration::Current, ActiveRecord::Migration, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, ActiveSupport::Tryable, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, Kernel, BasicObject]
irb(ActiveRecord::ConnectionAdapters::PostgreSQL::Migration::Compatibility::V5_0):002:0>
```